templates/bambooagent.service.j2

```diff
@@ -5,6 +5,13 @@ After=syslog.target network.target
 [Service]
 Environment=PATH=/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/bin:/bin{% if bamboo_agent is defined and bamboo_agent.capabilities is defined %}{% for item in bamboo_agent.capabilities %}{% if item.binary_path is defined %}:{{ item.binary_path }}{% endif %}{% endfor %}{% endif %}
 
+{%if bamboo_agent.use_proxy is defined %}
+Environment=JAVA_OPTS=-Dhttp.nonProxyHosts={% for item in bamboo_agent.no_proxy %}{{ item }}\|{% endfor %} -Dhttp.proxyHost={{ bamboo_agent.http_proxy_host }} -Dhttp.proxyPort={{ bamboo_agent.http_proxy_port }} -Dhttps.proxyHost={{ bamboo_agent.https_proxy_host }} -Dhttps.proxyPort={{ bamboo_agent.https_proxy_port }} -Dhttp.proxyUser={{ bamboo_agent.proxy_user }} -Dhttp.proxyPassword={{ bamboo_agent.proxy_password }}
+Environment=http_proxy=http://{{ proxy_user}}:{{ proxy_password }}@{{ proxy_host }}:{{ proxy_port }}
+Environment=https_proxy=http://{{ proxy_user}}:{{ proxy_password }}@{{ proxy_host }}:{{ proxy_port }}
+Environment=no_proxy={% for item in bamboo_agent.no_proxy %}{{ item }},{% endfor %}
+{% endif %}
+
```

Comment:
Why do you have HTTP proxy variable two times, like `bamboo_agent.http_proxy_host` and `proxy_host`?

Can we change the approach a little bit:
1. Don't use a boolean `use_proxy`, instead simply add the proxy if the variable is defined.
1. Harmonize variables with [ansible-role-bamboo](https://github.com/mimacom/ansible-role-bamboo/blob/master/templates/bamboo.service.j2), I suggest to use `http_proxy`and `https_proxy` dictionaries (no element under `bamboo_agent`)
1. Use `join()` filter instead of `for`

Maybe like this:
```
{%if http_proxy is defined and https_proxy is defined %}
Environment="JAVA_OPTS=-Dhttp.nonProxyHosts={{ http_proxy.no_proxy | default([]) | join("|") }} -Dhttp.proxyHost={{ http_proxy.host | default('') }} -Dhttp.proxyPort={{ http_proxy.port | default('') }} -Dhttps.proxyHost={{ https_proxy.host | default('') }} -Dhttps.proxyPort={{ https_proxy.port | default('') }} -Dhttp.proxyUser={{ http_proxy.user | default('') }} -Dhttp.proxyPassword={{ http_proxy.password | default('') }}"

Environment=http_proxy=http://{% if http_proxy.user is defined and http_proxy.password is defined %}{{ http_proxy.user }}:{{ http_proxy.password }}@{% endif %}{{ http_proxy.host }}:{{ http_proxy.port }}
Environment=https_proxy=http://{% if http_proxy.user is defined and http_proxy.password is defined %}{{ http_proxy.user }}:{{ http_proxy.password }}@{% endif %}{{ https_proxy.host }}:{{ https_proxy.port }}

Environment=no_proxy={{ http_proxy.no_proxy | default([]) | join(",") }}
{% endif %}
```

Tested with following data:
```
http_proxy:
  host: proxy.company.example
  port: 8080
  user: proxyusername
  password: pr0xyp4$$w0rd
  no_proxy:
    - "127.*"
    - "localhost"
    - "*.local.company.example"
https_proxy:
  host: proxy.company.example
  port: 8080
```

See: http://jinja2test.tk/